File: coletor/ConnectionFactory.py
import psycopg2

class ConnectionFactory :

    def getConnection(self):
        try:
            connection = psycopg2.connect(user="sys",
                                          password="sys",
                                          host="127.0.0.1",
                                          port="5432",
                                          database="memch")

            cursor = connection.cursor()
            # Print PostgreSQL Connection properties
            #print(connection.get_dsn_parameters(), "\n")

            return connection
        except (Exception, psycopg2.Error) as error:
            print("Error while connecting to PostgreSQL", error)
        #finally:
        #    # closing database connection.
        #    if (connection):
        #        cursor.close()
        #        connection.close()
        #        print("PostgreSQL connection is closed")

File: servidor/src/java/util/ConnectionFactory.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package util;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.logging.Level;
import java.util.logging.Logger;

/**
 *
 * @author osmar
 */
public class ConnectionFactory {

    private final String DRIVER = "org.postgresql.Driver";
    private final String URL = "jdbc:postgresql://127.0.0.1:5432/memch";
    private final String USER = "sys";
    private final String PASS = "sys";

    public Connection getConnection() {
        try {
            Class.forName(DRIVER);

            return DriverManager.getConnection(URL, USER, PASS);
        } catch (ClassNotFoundException | SQLException ex) {
            throw new RuntimeException("Erro na Conexão: ", ex);
        }
    }

    public static void closeConnection(Connection con) {

        try {
            if (con != null) {
                con.close();
            }
        } catch (SQLException ex) {
            Logger.getLogger(ConnectionFactory.class.getName()).log(Level.SEVERE, null, ex);
        }
    }
    public static void closeConnection(Connection con, PreparedStatement stmt) {

        closeConnection(con);
        
        try {
            if (stmt != null) {
                stmt.close();
            }
        } catch (SQLException ex) {
            Logger.getLogger(ConnectionFactory.class.getName()).log(Level.SEVERE, null, ex);
        }
    }
    
    
    public static void closeConnection(Connection con, PreparedStatement stmt, ResultSet rs) {

        closeConnection(con,stmt);
        
        try {
            if (rs != null) {
                rs.close();
            }
        } catch (SQLException ex) {
            Logger.getLogger(ConnectionFactory.class.getName()).log(Level.SEVERE, null, ex);
        }
    }
}
